GuessingGame/Service.cs

```diff
@@ -3,71 +3,75 @@
 
 namespace GuessingGame
 {
-public class Service
-{
-     private readonly INumberGenerator _numberGenerator;
+    public class GuessingGameService
+    {
+        private readonly INumberGenerator _numberGenerator;
 
-        public Service(INumberGenerator numberGenerator)
+        public GuessingGameService(INumberGenerator numberGenerator)
         {
             _numberGenerator = numberGenerator;
         }
-        
 
-        public int GetAmount()
-    {
-    Console.WriteLine("Enter the amount of money to bet:");
-     if(!int.TryParse(Console.ReadLine(),out int amount)){
-        Console.WriteLine("Invalid input. Please enter a valid whole number.");
-        GetAmount();
-     }
-     return amount;
-     }
 
-    public int GetNumber()
-    {
-        Console.Write("Enter the number to guess: ");
-    if(!int.TryParse(Console.ReadLine(),out int number)){
-        Console.WriteLine("Invalid input. Please enter a valid whole number.");
-    }
+        public int MoneyToBet()
+        {
+            Console.WriteLine("Enter the amount of money to bet:");
 
-    return number;
-    }
+            if (!int.TryParse(Console.ReadLine(), out int amount))
+            {
+                Console.WriteLine("Invalid input. Please enter a valid whole number.");
+                MoneyToBet();
+            }
+            return amount;
+        }
+
+        public int GuessNumber()
+        {
+            Console.Write("Enter the number to guess: ");
+            if (!int.TryParse(Console.ReadLine(), out int number))
+            {
+                Console.WriteLine("Invalid input. Please enter a valid whole number.");
+            }
 
-    public string GetLevel(){
-        Console.Write(" Choose level (Easy/Medium/Hard) : ");
-        string difficulty = Console.ReadLine();
-        
-                    switch (difficulty.ToLower())
-                    {
-                    case "easy":
-                    return GetResult(1,5,5);
-                    case "medium":
-                    return GetResult(1,10,10);  
-                    case "hard":
-                    return GetResult(1,15,15);
-                    default:
+            return number;
+        }
+
+        public string ChooseDifficultyLevelWhichReturnMinMaxMutipler(int amount,int guessNumber)
```

Comment:
The method name here is slightly better, for the Lambda exersise, it would be good to start considering Single Responsibility principle for methods, This method is starting to do too many things which makes it difficult to name.